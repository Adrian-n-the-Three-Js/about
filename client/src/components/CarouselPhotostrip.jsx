import React from 'react';
import styled from 'styled-components';

console.log('album toggle');

const CarouselPhotostrip = (props) => {
  return (
    <Image className="photostrip"
      alt={props.caption}
      src={props.photo}
      onClick={ () => props.onClick(props.index) }
    />
  );
};

const Image = styled.img`
  flex: 1;
  max-height: 100px
  // position: relative;
  // margin: 1px;
  // flex-shrink: 1;
  // padding: 0;
  // // width: 100%;
  // height: 50px;
  // display: block;
`;

export default CarouselPhotostrip;
