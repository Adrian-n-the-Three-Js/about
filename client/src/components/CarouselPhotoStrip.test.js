import React from 'react';
import { mount, shallow, render } from 'enzyme';
import CarouselPhotostrip from './CarouselPhotostrip.jsx';

describe ('CarouselPhotostrip', () => {
  it('should render correctly', () => {
    const PhotostripComponent = shallow(<CarouselPhotostrip />);
    expect(PhotostripComponent).toMatchSnapshot();
  });

  it('should trigger onClick prop when clicked', () => {
    const mock = jest.fn();
    const PhotostripComponent = shallow(<CarouselPhotostrip onClick={mock} />);
    PhotostripComponent.find('.photostrip').simulate('click');
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
