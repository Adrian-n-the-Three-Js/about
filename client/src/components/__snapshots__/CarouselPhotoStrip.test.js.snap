// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselPhotostrip should render correctly 1`] = `
<CarouselPhotostrip__Image
  className="photostrip"
  onClick={[Function]}
/>
`;
